manta-pay/src/test/ledger.rs

```diff
@@ -0,0 +1,598 @@
+                rng.seed_rng().expect("Failed to sample PRNG for signer."),
+            ),
+        )
+    }
+
+    /// Runs a simple simulation to test that the signer-wallet-ledger connection works.
+    #[async_std::test]
+    async fn test_simulation() {
+        let directory = task::spawn_blocking(tempfile::tempdir)
+            .await
+            .expect("Unable to generate temporary test directory.");
+        println!("[INFO] Temporary Directory: {:?}", directory).await;
+
+        let mut rng = thread_rng();
+        let parameters = rng.gen();
+        let utxo_set_parameters = rng.gen();
+
+        let (proving_context, verifying_context) = transfer::canonical::generate_context(
+            &(),
+            FullParameters::new(&parameters, &utxo_set_parameters),
+            &mut rng,
+        )
+        .expect("Failed to generate contexts.");
+
+        let cache = OnDiskMultiProvingContext::new(directory.path());
+        cache
+            .save(proving_context)
+            .await
+            .expect("Unable to save proving context to disk.");
+
+        const ACTOR_COUNT: usize = 10;
+
+        let mut ledger = Ledger::new(utxo_set_parameters.clone(), verifying_context);
+
+        for i in 0..ACTOR_COUNT {
+            ledger.set_public_balance(AccountId(i as u64), AssetId(0), AssetValue(1000000));
+            ledger.set_public_balance(AccountId(i as u64), AssetId(1), AssetValue(1000000));
+            ledger.set_public_balance(AccountId(i as u64), AssetId(2), AssetValue(1000000));
+        }
+
+        let ledger = Rc::new(RwLock::new(ledger));
+
+        println!("[INFO] Building Wallets").await;
+
+        let actors = (0..ACTOR_COUNT)
+            .map(|i| {
+                Actor::new(
+                    sample_wallet(
+                        AccountId(i as u64),
+                        &ledger,
+                        &cache,
+                        &parameters,
+                        &utxo_set_parameters,
+                        &mut rng,
+                    ),
+                    Default::default(),
+                    rand::Rng::gen_range(&mut rng, 50..300),
+                )
+            })
+            .collect::<Vec<_>>();
+
+        let mut simulator = Simulator::new(ActionSim(Simulation::default()), actors);
+
+        println!("[INFO] Running Simulation\n").await;
+
+        let mut events = simulator.run(thread_rng);
+        while let Some(event) = events.next().await {
+            match event.event.action {
+                ActionType::Skip | ActionType::GeneratePublicKey => {}
+                _ => println!("{:?}", event).await,
+            }
+            if let Err(err) = event.event.result {
+                println!("\n[ERROR] Simulation Error: {:?}\n", err).await;
+                break;
+            }
+        }
+
+        task::spawn_blocking(move || directory.close())
+            .await
+            .expect("Unable to delete temporary test directory.");
+    }
+}
```

Comment:
Want to add a check that account-level balances are accurate, as we do in manta-js now